lib/ain-evm/src/core.rs

```diff
@@ -816,6 +816,91 @@ impl EVMCoreService {
 
         Ok((listener.trace, execution_success, return_value, used_gas))
     }
+
+    pub fn create_access_list(&self, arguments: EthCallArgs) -> Result<(AccessList, u64)> {
+        let EthCallArgs {
+            caller,
+            to,
+            value,
+            data,
+            gas_limit,
+            access_list,
+            block_number,
+            ..
+        } = arguments;
+
+        let block_header = self
+            .storage
+            .get_block_by_number(&block_number)?
+            .ok_or_else(|| format_err!("Block not found"))
+            .map(|block| block.header)?;
+        let state_root = block_header.state_root;
+        debug!(
+            "Calling EVM at block number : {:#x}, state_root : {:#x}",
+            block_number, state_root
+        );
+
+        let mut vicinity = Vicinity::from(block_header);
+        vicinity.origin = caller;
+        vicinity.gas_price = arguments.gas_price;
+
+        let backend = EVMBackend::from_root(
+            state_root,
+            Arc::clone(&self.trie_store),
+            Arc::clone(&self.storage),
+            vicinity,
+            None,
+        )
+        .map_err(|e| format_err!("Could not restore backend {}", e))?;
+
+        static CONFIG: Config = Config::shanghai();
+        let metadata = StackSubstateMetadata::new(gas_limit, &CONFIG);
+        let state = MemoryStackState::new(metadata.clone(), &backend);
+        let al_state = MemoryStackState::new(metadata, &backend);
+        let precompiles = MetachainPrecompiles;
+        let mut executor = StackExecutor::new_with_precompiles(state, &CONFIG, &precompiles);
+        let mut al_executor = StackExecutor::new_with_precompiles(al_state, &CONFIG, &precompiles);
+
+        let mut listener = crate::eventlistener::StorageAccessListener::new();
+
+        runtime_using(&mut listener, move || {
+            let access_list = access_list
+                .into_iter()
+                .map(|x| (x.address, x.storage_keys))
+                .collect::<Vec<_>>();
+
+            executor.transact_call(
+                caller,
+                to.unwrap(),
+                value,
+                data.to_vec(),
+                gas_limit,
+                access_list,
+            );
+        });
+
+        let al: AccessList = listener
+            .access_list
+            .into_iter()
+            .map(|(address, storage_keys)| AccessListItem {
+                address,
+                storage_keys: Vec::from_iter(storage_keys),
+            })
+            .collect();
+
+        al_executor.transact_call(
```

Comment:
Why are we executing another transaction call here while not doing anything with the execution results?

Reply:
https://github.com/DeFiCh/ain/blob/04c36dc6d08082d1d60bc0822a70151213580df4/lib/ain-evm/src/core.rs#L924

It is part of the results returned to the user.

Reply:
Likewise here